mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/CompareWithCurrentThreshold.java

```diff
@@ -0,0 +1,27 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.quota;
+
+public enum CompareWithCurrentThreshold {
```

Comment:
An result that reflects a result Enum can't be a verb.
What about ThresholdChange ?
Also, Current is ambiguous, what about HigherThresholdReached, NoChange, LowerThresholdReached ?
Finally, you are mixing two things here by trying to handle recentness, I think you should not.

Reply:
True. I did two separate enum and a POJO for this.

It also allows extracting some logic from  other classes.